### PR TITLE
Update configuration file documentation and some miscellany

### PR DESCRIPTION
**Pull request by mfisher87**
There's some more work to do before this is ready to merge. I felt that there was a need for some examples around making basic customizations to the i3 config. The new way configs are installed threw me for a loop and I tried to document things that I learned and felt were important in this PR.

For the more major changes, I tried to preserve your wording, but in some cases moved sections to different files to keep things more focused where I added more content that I felt would otherwise be overwhelming.

I noticed you have a "Reference" and "How To" section. Are you using [Diataxis](https://diataxis.fr/) to write these docs? I was thinking it may be worth extracting some content to a "Discussions" section.

I gave the new "customizing i3" how-to a weight so it would go to the top of the list, because I felt it was a key beginner how-to. 

The new how-to on customizing the compositor links to 1.6 docs only, but I felt something should show up when you type "compositor" into the search on this site. For some reason, this did _not_ achieve that. I'm confused about how the search bar works. Anyway, regarding content for this page: Has the default compositor changed? I noticed that I have the `none` compositor package installed on my main workstation, but `picom` is in fact running. Maybe an artifact of upgrading from 1.6? Anyway, someone else on Slack today had noticed the same thing: that the `none` compositor package was installed by default. But they also had no _running_ compositor, so not exactly the same thing.

I added a docker-compose configuration and relevant entry in README -- I felt this was an easier way for me to get started working today, so why not commit the effort in case it helps someone else contribute? I haven't used Hugo before, but I have used NPM for JS development. I just didn't feel like setting up NPM on this machine.

This is a rambly and unfocused comment on a rambly and unfocused PR... please let me know what you'd like me to change, or feel free to do what you will with my input :)

### CODE REVIEW

**Comment by mfisher87** [comment]
- [ ] Run prettier

I aligned the tables for my own readability, and I'm not sure if `prettier` is going to set it back. We'll see!

**Comment by mfisher87** [comment]
- [ ] Cleanup TODOs or ephemeral NOTEs in doc pages

File: README.md [hunk 1]
```diff
@@ -12,13 +12,32 @@ Please ensure you have the following installed:
 
 ### Checkout and View Local
 
+Clone this repo, and the Book theme as a submodule:
+
 ```bash
```
Comment (mfisher87) [inline README.md:17]: The README mentioned not to use the `bash` code annotation and use `console` instead, but this was pre-existing. Wasn't sure if there was a good reason for it or if I should have updated it :)

File: content/docs/using-regolith/configuration.md [hunk 2]
```diff
@@ -188,7 +181,7 @@ i3-wm.mod: Mod1
 i3-wm.alt: Mod4
 ```
```
Comment (mfisher87) [inline content/docs/using-regolith/configuration.md:182]: I'm not sure why github is highlighting these :point_up: in red... they are still there according to my local copy.

File: content/docs/howtos/customize-i3-configuration.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+```console
+dpkg -S /usr/share/doc/regolith-i3-workspace-config
+```
+
+The output contains the name of the package:
+
+```
+regolith-i3-workspace-config: /usr/share/doc/regolith-i3-workspace-config
+```
```
Comment (mfisher87) [inline content/docs/howtos/customize-i3-configuration.md:63]: I admit I didn't look very hard for other examples, but what's the convention in this documentation for documenting a command and its output?

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    #
+    #    fatal: unsafe repository ('/src' is owned by someone else)
+    #
+    # This is because Git 2.35.2 added this check as a security fix. The
+    # non-minimal image has an older version of Git, so it doesn't complain.
+    # Can be fixed with a .gitconfig containing:
+    #
+    #     [safe]
+    #         directory = /home/repon
+    image: "klakegg/hugo:0.101.0-ext"
```
Comment (kgilmer) [inline docker-compose.yml:19]: question: I do not use docker compose so am wondering, how stable is this image uri?  Do we expect this to continue to work, say, next year?  If not is there another URI form that would map to "latest" or something?

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+version: '3.4'
```
Comment (kgilmer) [inline docker-compose.yml:1]: Nice, thanks for adding this!

**Review by kgilmer — APPROVED** [review]
Looks great, I've responded to a few WIP items here but otherwise LGTM

**Comment by kgilmer** [comment]
>  I noticed that I have the none compositor package installed on my main workstation, but picom is in fact running. Maybe an artifact of upgrading from 1.6?

Interesting, it's unclear to me how picom loading would "survive" an upgrade without the containing package to be installed.  Each compositor is mutually exclusive so unsure of how your system got into that state.

**Comment by mfisher87** [comment]
Thanks for your comments! I'll get back to them soon. I don't have the energy after working today :)

File: content/docs/using-regolith/configuration.md (hunk 2 above)
Comment (kgilmer) [reply content/docs/using-regolith/configuration.md:182]: I think there is just some markdown parsing rule that's being tripped unintentionally.  I don't think it matters.

File: README.md (hunk 1 above)
Comment (mfisher87) [reply README.md]: :heavy_check_mark: 392fbfb

File: content/docs/using-regolith/configuration.md (hunk 2 above)
Comment (mfisher87) [reply content/docs/using-regolith/configuration.md:182]: I wish GitHub hadn't chosen the same color as a "deletion" to highlight the parsing issue :laughing:

File: content/docs/howtos/customize-i3-configuration.md (hunk 3 above)
Comment (mfisher87) [reply content/docs/howtos/customize-i3-configuration.md:63]: I.e. prefix the commands with a `$` and include the output in same code block like:

```
$ command
the output of the command
...
more important output
```

?

File: content/docs/howtos/customize-compositor.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+title: "Customize Compositor"
+linkTitle: "How to Customize Compositor"
+description: >
+  How to customize Regolith's compositor
+---
+
+{{< hint warning >}}
+This page is under construction
+
+TODO: What is the default compositor in Regolith 2.0?
```
[the remark this answers is not in the document]
Comment (mfisher87) [reply content/docs/howtos/customize-compositor.md:11]: Great, thanks for that info :) I wanted to call out that change in default compositors between 1.x and 2.x. That sounds like a great reason for the change; I'll document this reasoning as well!

Would you like me to call out NVIDIA on this page somehow? Maybe an NVIDIA-specific compositor rec?

FWIW I'm personally using NVIDIA hardware and picom works well (although, as mentioned in another comment in this PR, I have `regolith-compositor-none` installed. I must have done something weird before upgrading to Regolith 2.x). In the 1.x version of Regolith, I found myself needing to restart picom often. I even [threw together a little script for it](https://github.com/mfisher87/picom-ensure/blob/main/picom-ensure), and bound it to `mod`+`p`. In 2.x, either config or a new version of picom itself seems to have resolved that issue and it looks like picom is "just running" all the time now.

File: content/docs/howtos/customize-i3-configuration.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+set_from_resource $i3-wm.program.compositor i3-wm.program.compositor /usr/share/regolith-compositor/init
+exec_always --no-startup-id $i3-wm.program.compositor
+```
+
+To do this, simply add a line to `~/.config/regolith2/Xresources` like:
+
+```
+# Use my own home-made compositor
+i3-wm.program.compositor: <my-custom-compositor>
```
[the remark this answers is not in the document]
Comment (mfisher87) [reply content/docs/howtos/customize-i3-configuration.md]: :heavy_check_mark: deee624

File: content/docs/howtos/customize-i3-configuration.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+Remove the partial:
+
+```console
+sudo apt remove regolith-i3-workspace-config
+```
+
+## How to customize a default partial
+
+Ensure that a directory exists for user config partials. *NOTE*: It's important that the file `~/.config/regolith2/i3/config` does **not** exist.
```
[the remark this answers is not in the document]
Comment (mfisher87) [reply content/docs/howtos/customize-i3-configuration.md:79]: Good call!

[the remark this answers is not in the document]
Comment (mfisher87) [reply content/docs/howtos/customize-i3-configuration.md]: :heavy_check_mark: f7f8863 and fad1d4d

File: docker-compose.yml (hunk 4 above)
Comment (mfisher87) [reply docker-compose.yml:19]: Good call, Docker image tagging conventions do have a concept of `latest`. Tags are free-form like git tags. In this repo, `latest` itself corresponds to a non-"extended" version of Hugo, but we want "extended". The README for this image repository (https://hub.docker.com/r/klakegg/hugo/) describes a `latest-ext` tag that looks exactly like what we want.

Comment (mfisher87) [reply docker-compose.yml]: :heavy_check_mark: 96711dd

Comment (mfisher87) [reply docker-compose.yml]: Forgot to answer one of your questions:

> Do we expect this to continue to work, say, next year?

I would expect `0.101.0-ext` to continue to work as long as someone is pulling that version of the image from DockerHub every month or so. I forget the exact policy, but if you don't pay DockerHub, they will archive your images that haven't been pulled in a certain amount of time. A `latest` image is definitely safer against this possibility.

File: content/docs/howtos/customize-compositor.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (mfisher87) [reply content/docs/howtos/customize-compositor.md:11]: - [ ] compositor documentation

**Comment by mfisher87** [comment]
Have you had a chance to look at Diataxis? [This YouTube video (A PyCon AU 2017 talk)](https://www.youtube.com/watch?v=t4vKPhjcMZg) is my favorite intro to Diataxis (the video was made before the project had a name). For some reason, the video was removed from the main Diataxis website. Perhaps due to the naming thing!

**Comment by mfisher87** [comment]
>Interesting, it's unclear to me how picom loading would "survive" an upgrade without the containing package to be installed. Each compositor is mutually exclusive so unsure of how your system got into that state.

I'm also unsure, except that I'm _pretty_ sure it's something weird I did :) May not be reproducible for others.

File: content/docs/howtos/customize-compositor.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (mfisher87) [reply content/docs/howtos/customize-compositor.md:11]: I threw together the compositor doc as a separate PR so make review easier. I've never tried to do a cross-fork PR that depends on another cross-fork PR so for now this is just against my own fork, but once this is merged, I'll re-target it.
https://github.com/mfisher87/regolith-desktop-website/pull/1

**Comment by mfisher87** [comment]
I do some JS development, mostly Python, but haven't used Prettier before. I use Black in Python land, which seems similar. Anyway, I was looking at the docs for prettier and saw:

```
Note: If you forget to install Prettier first, npx will temporarily download the latest version. That’s not a good idea when using Prettier, because we change how code is formatted in each release! It’s important to have a locked down version of Prettier in your package.json. And it’s faster, too.
```

Maybe this is another PR :) I'll track it with a new issue.

**Comment by mfisher87** [comment]
This PR resolves #5 . Will it be automatically closed because of this comment, or should I have added `resolves #5` to a commit? :shrug:

File: content/docs/reference/configurations.md [hunk 9]
```diff
@@ -6,13 +6,35 @@ description: >
...
-NOTE: Adding an i3 config file in `~/.config/regolith2/i3/config` will essentially override all Regolith configuration including Xresources and looks.
-{{< /hint >}}
\ No newline at end of file
+WARNING: Adding an i3 config file in `~/.config/regolith2/i3/config` will essentially override all Regolith configuration including Xresources and looks.
+{{< /hint >}}
+
+# History
+
+{{< hint info >}}
+NOTE: Under the Diataxis documentation framework, I think this would be more of a "discussion". Is this site using Diataxis?
```
[the remark this answers is not in the document]
Comment (mfisher87) [reply content/docs/reference/configurations.md:25]: I think if we go that route, let's put it in a dedicated PR :) this one is already spread pretty thin! I'll clean that note up and resolve this.

[the remark this answers is not in the document]
Comment (mfisher87) [reply content/docs/reference/configurations.md]: :heavy_check_mark: 4fa7b3e

**Comment by mfisher87** [comment]
@kgilmer I had a few questions, and extracted a new issue #9 and new PR mfisher87#1 from discussions here. Once this is merged, I'll open the next PR for the compositor docs!

File: content/docs/howtos/customize-compositor.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (mfisher87) [reply content/docs/howtos/customize-compositor.md]: I've opened a PR to address what I think is the cause of the compositor issue described above. https://github.com/regolith-linux/regolith-look-extra/pull/8

File: content/docs/howtos/customize-i3-configuration.md (hunk 3 above)
Comment (kgilmer) [reply content/docs/howtos/customize-i3-configuration.md:64]: This has been a point of contention in the past.  People seem to prefer w/out the `$` prefix.  Good as is.

**Review by kgilmer — APPROVED** [review]
LGTM

**Comment by kgilmer** [comment]
Thanks @mfisher87 , big improvement for the i3 docs!

**Pull request merged**